### PR TITLE
Bps simulator enhacements

### PR DESCRIPTION
**Pull request by SoroushF79**
Simulator now has checkbox for charging. TEST_CAN leads to stack smashing.

### CODE REVIEW

**Comment by SoroushF79** [comment]
I forgot that the id shows up in decimal not hex but I couldn't find the problem for that

File: BSP/Simulator/Hardware/simulate.py [hunk 1]
```diff
@@ -87,6 +89,9 @@ def update_lights():
         light.set(f"{lights[i]}: {(lights_state >> i) & 0x01}")
     window.after(LIGHTS_FREQ, update_lights)
 
+def update_MotorDisable():
+    """Sends MOTOR_DISABLE message when checkbox is checked/unchecked"""
+    MotorMessage.sendMotorDisable(chargingBool.get())
```
Comment (JimothyGreene) [inline BSP/Simulator/Hardware/simulate.py:94]: Make sure that you add a call to the `charging_frame.after()` method here so it calls this method periodically. Right now it's only calling this method when the checkmark is pressed. We want the BPS message to be continuously broadcasted on a periodic interval all the time. The only thing that changes is the data packet.

**Review by JimothyGreene — CHANGES_REQUESTED** [review]
See comments

**Review by JimothyGreene — CHANGES_REQUESTED** [review]
Looking better now. Can you get it to display the ID in hex? You can use the Python built-in method `hex()` to get the hex string of an integer

**Review by JimothyGreene — APPROVED** [review]
Looks good to me

**Review by Cam0Cow — APPROVED** [review]

**Pull request merged**